napari/utils/transforms/transforms.py

```diff
@@ -148,12 +160,31 @@ def __getitem__(self, key: slice) -> 'TransformChain[_T]':
         ...
 
     def __getitem__(self, value):
-        return super().__getitem__(value)
+        if f"getitem_{value}" not in self._cache_dict:
```

Comment:
This isn't new, but I'd prefer if the definition of this type-overloaded method uses `key` instead of `value` for consistency.

That also helps a little with the changes here because when I first saw value, I though of transform values (which would not be appropriate to key on for this cache) rather than keys for the chain (which should be `int`, `str`, or `slice`, all of which are appropriate for this cache key).

Reply:
done